src/status_im/transport/filters.cljs

```diff
@@ -0,0 +1,70 @@
+(ns status-im.transport.filters
+  (:require [re-frame.core :as re-frame]
+            [status-im.utils.handlers :as handlers]
+            [status-im.transport.utils :as utils]
+            [status-im.utils.config :as config]
+            [taoensso.timbre :as log]))
+
+(defn remove-filter! [filter]
+  (.stopWatching filter
+                 (fn [error _]
+                   (when error
+                     (log/warn :remove-filter-error filter error))))
+  (log/debug :stop-watching filter))
+
+(defn add-shh-filter!
+  [web3 options callback]
+  (.newMessageFilter (utils/shh web3) (clj->js options)
+                     callback
+                     #(log/warn :add-filter-error (.stringify js/JSON (clj->js options)) %)))
+
+(defn add-filter!
+  [web3 {:keys [topics to] :as options} callback]
+  (let [options  (if config/offline-inbox-enabled?
+                   (assoc options :allowP2P true)
+                   options)]
+    (log/debug :add-filter options)
+    (add-shh-filter! web3 options callback)))
+
+(re-frame/reg-fx
```

Comment:
Is add-filter! and remove-filter! part of this ns API? I.e. should they be public or not? Not clear to me since add-filter is fx. If public, docstrings.

Reply:
nothing is really private in clojure, but normally we should use events and fx and not functions directly